### DISCUSSION TITLE
[Parameter #0 [ <required> array $config ]] in class Spatie\LaravelData\Support\DataConfig while testing

### DISCUSSION THREAD

**Discussion by elabuwa** [spatie/laravel-data#469] [Q&A] [answered]
I've got the data object set up as below.

```
<?php

namespace Instance\Data;

use Carbon\Carbon;
use Spatie\LaravelData\Data;

class ReportingPeriodData extends Data
{
    public function __construct(
      public Carbon $fromDate,
      public Carbon $toDate,
    ) {}
}
```

I've got the test written as below. I've stripped everything just to have what's needed for this post, but even the below still return the error.

```
<?php

namespace Tests\Unit\Instance;

use Carbon\Carbon;
use Instance\Data\ReportingPeriodData;
use PHPUnit\Framework\TestCase;


class TestTransactionDateAllowPeriodTest extends TestCase
{
    public function test_is_in_allowed_period(): void
    {
        $reportingData = ReportingPeriodData::from([
            'fromDate' => Carbon::parse('2025-04-01'),
            'toDate' => Carbon::parse('2026-03-31')
        ]);
    }
}

```
However, I get the below error.

>  Unresolvable dependency resolving [Parameter #0 [ <required> array $config ]] in class Spatie\LaravelData\Support\DataConfig
> 
>  at vendor/laravel/framework/src/Illuminate/Container/Container.php:1143
>     1139▕     protected function unresolvablePrimitive(ReflectionParameter $parameter)
>     1140▕     {
>     1141▕         $message = "Unresolvable dependency resolving [$parameter] in class {$parameter->getDeclaringClass()->getName()}";
>     1142▕ 
>   ➜ 1143▕         throw new BindingResolutionException($message);
>     1144▕     }
>     1145▕ 
>     1146▕     /**
>     1147▕      * Register a new before resolving callback for all types.
> 
>       +13 vendor frames 
>   14  tests/Unit/Instance/TestTransactionDateAllowPeriodTest.php:15

The code runs as expected when a request comes through. 
But this errors out simply during creating the first test.

Any hep is greatly appreciated.

Versions
spatie/laravel-data : 3.5.1
laravel/framework : 10.6.1
php : 8.1.14

Also, I've published the config file as well.

There is another thread where the same issue has occured, but it doesn't mention any solutions.

**Comment by elabuwa** [comment, marked as answer]
I've resolved this.
Turns out, you need to extend from "Tests\TestCase" rather than "PHPUnit\Framework\TestCase".

**Reply by edalzell** [reply]
I'm getting this now in a package so I can't extend that, what is the actual cause of this?

**Reply by edalzell** [reply]
I needed to add Laravel Data's Service Provider to my `getPackageProviders`
